test/e2e/scrapeconfig_test.go

```diff
@@ -2119,3 +2122,148 @@ var LightSailSDTestCases = []scrapeCRDTestCase{
 		expectedError: true,
 	},
 }
+
+var OVHCloudSDTestCases = []scrapeCRDTestCase{
```

Comment:
(nit) Try grouping together test cases related to a given field, to make it easier to read. For example, the test cases which cover a correctly defined Service, a wrongful Service, a missing Service etc should all be next to each other. This way we can quickly see what test cases are missing, if something is redundant etc.

Reply:
In general, 
we need test cases where the field is defined correctly, defined incorrectly (this includes wrong texts, empty texts, etc), and not defined at all.

For example, you have added a test case for empty (incorrectly defined) and correctly defined `ApplicationKey`, but seem to have missed a test case for "missing `ApplicationKey`". It would be really easy to catch if you group the tests together.